examples/util/openvino/OpenVINOModelImporter.js

```diff
@@ -620,15 +626,65 @@ class OpenVINOModelImporter {
           opCode = this._nn.CONCATENATION;
         } break;
         case 'Permute': {
+          const input = node.inputs[0];
           const order = node.getInts('order');
+          const inDims = input.shape();
+          const inputId = this._getTensorId(input);
+          const output = node.outputs[0];
+          const outputName = output.graphId();
           if (order.toString() === '0,2,3,1') {
-            const inputId = this._getTensorId(node.inputs[0]);
-            const outputName = node.outputs[0].graphId();
             this._tensorIds[outputName] = inputId;
             // equivalent to NCHW -> NHWC
-            console.log(`  skip permuting to [0, 2, 3, 1]`);
+            console.log(`  skip permuting to ${order.toString()}`);
           } else {
-            throw new Error(`Permuting to ${order} is not supported`);
+            if (order.length === 4) {
+              console.log(`  input shape: [${inDims}]`);
+
+              // step 1: NHWC -> NCWH
+              inputs.push(inputId);
+              inputs.push(this._addTensorInt32([0, 3, 1, 2], [4]));
+
+              const step1OutputDims = [inDims[0], inDims[3], inDims[1], inDims[2]];
+              const step1OutputType = {type: this._nn.TENSOR_FLOAT32, dimensions: step1OutputDims};
+              const step1OutputId = this._addOperand(step1OutputType);
+
+              outputs.push(step1OutputId);
+              this._addOperation(this._nn.TRANSPOSE, inputs, outputs);
+
+              // step 2: transpose to order
+              inputs = [];
+              outputs = [];
+              inputs.push(step1OutputId);
+              inputs.push(this._addTensorInt32(order, [order.length]));
+
+              let step2OutputDims = [];
+              for (let number in order) {
+                step2OutputDims[number] = step1OutputDims[order[number]];
+              }
+              const step2OutputType = {type: this._nn.TENSOR_FLOAT32, dimensions: step2OutputDims};
+              const step2OutputId = this._addOperand(step2OutputType);
+
+              outputs.push(step2OutputId);
+              this._addOperation(this._nn.TRANSPOSE, inputs, outputs);
+
+              // step 3: NCWH -> NHWC
+              inputs = [];
+              outputs = [];
+              inputs.push(step2OutputId);
+              inputs.push(this._addTensorInt32([0, 2, 3, 1], [4]));
+
+              const outDims = output.shape();
+              const outputType = {
+                type: this._getTypeCode(output.dataType()), dimensions: outDims
+              };
+              const outputId = this._addNamedOperand(outputName, outputType);
+              outputs.push(outputId);
+              console.log(`  output shape: [${outDims}]`);
+
+              this._addOperation(this._nn.TRANSPOSE, inputs, outputs);
```

Comment:
Why do we need 3 `TRANSPOSE` ops? Can we map the axis between NHWC and NCHW and use only one transpose?

Reply:
@huningxin Yes, we can. I transpose `input` data, but you want to transpose `order` data and use only one op. So I will update this op.